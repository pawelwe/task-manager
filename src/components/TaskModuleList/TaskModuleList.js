import React from 'react';
import TaskModule from './TaskModule/TaskModule';
import classes from './TaskModuleList.css';

const TaskModuleList = ({
  taskModules,
  addNewTask,
  toggleTask,
  removeTask,
  removeTaskModule,
  sortByPriority,
  sortByExpiration,
}) => {
  return (
    <ul className={`${classes.TaskModuleList}`}>
      {taskModules.map(module => {
        return (
          <TaskModule
            moduleId={module.id}
            key={module.id}
            addNewTask={addNewTask}
            toggleTask={toggleTask}
            removeTask={removeTask}
            removeTaskModule={removeTaskModule}
            {...module}
            sortByPriority={sortByPriority}
            sortByExpiration={sortByExpiration}
          />
        );
      })}
    </ul>
  );
};

export default TaskModuleList;
