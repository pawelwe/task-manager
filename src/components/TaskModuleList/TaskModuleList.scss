
.TaskModuleList {
  display: flex;
  flex-wrap: wrap;
}