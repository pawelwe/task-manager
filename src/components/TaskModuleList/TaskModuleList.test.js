import React from 'react';
import { shallow } from 'enzyme';
import TaskModuleList from './TaskModuleList';

describe('TaskModuleList Component', () => {
  const wrapper = shallow(<TaskModuleList taskModules={[]} />);

  it('Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
