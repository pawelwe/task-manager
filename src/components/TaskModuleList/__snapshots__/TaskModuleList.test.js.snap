// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskModuleList Component Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskModuleList
    taskModules={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "undefined",
    },
    "ref": null,
    "rendered": Array [],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "undefined",
      },
      "ref": null,
      "rendered": Array [],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
